src/lua/api-gateway/dogstatsd/Dogstatsd.lua

```diff
@@ -0,0 +1,70 @@
+
+local Dogstatsd = {}
+
+function Dogstatsd:new(o)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    return o
+end
+
+--- Loads a lua gracefully. If the module doesn't exist the exception is caught, logged and the execution continues
+-- @param module path to the module to be loaded
+--
+local function loadrequire(module)
+    ngx.log(ngx.DEBUG, "Loading module [" .. tostring(module) .. "]")
+    local function requiref(module)
+        require(module)
+    end
+
+    local res, cls = pcall(requiref, module)
+    if res then
+        return cls
+    else
+        ngx.log(ngx.WARN, "Could not load module [", module, "].")
+        return nil
+    end
+end
+
+--- Returns an instance of dogstatsd only if it does not already exist
+function Dogstatsd:getDogstatsd()
```

Comment:
seems to me like this should not be on the Dogstatsd - it's a singleton pattern on a non singleton class - it's a singleton require though. So can be a private local function